generators/kubernetes/templates/registry/_jhipster-registry.yml

```diff
@@ -125,13 +137,13 @@ spec:
             LAST_POD_INDEX=$((${CLUSTER_SIZE} - 1))
             REPLICAS=""
             for i in $(seq 0 $LAST_POD_INDEX); do
-              REPLICAS="${REPLICAS}http://admin:${SECURITY_USER_PASSWORD}@${STATEFULSET_NAME}-${i}.jhipster-registry.${STATEFULSET_NAMESPACE}.svc.cluster.local:8761/eureka/"
+              REPLICAS="${REPLICAS}http://admin:${JHIPSTER_REGISTRY_PASSWORD}@${STATEFULSET_NAME}-${i}.jhipster-registry.${STATEFULSET_NAMESPACE}.svc.cluster.local:8761/eureka/"
               if [ $i -lt $LAST_POD_INDEX ]; then
                 REPLICAS="${REPLICAS},"
               fi
             done
-            export EUREKA_CLIENT_SERVICE_URL_DEFAULTZONE=$REPLICAS
-            echo "EUREKA_CLIENT_SERVICE_URL_DEFAULTZONE=${REPLICAS}"
+            export EUREKA_CLIENT_SERVICEURL_DEFAULTZONE=$REPLICAS
+            echo "EUREKA_CLIENT_SERVICEURL_DEFAULTZONE=${REPLICAS}"
```

Comment:
@xetys - Yeah, it needs to be changed. Since you have the same info in application.xml, it takes preference and hence it figures out the replicas. (if you remove the service-url entry from application.yml, then this would fail).

I am testing with this branch (xetys/fix-eureka-k8s), but still I see issues when you increase the replica count from the default one(2).  Try running with registerWithEureka 'true', you would be able to see the inconsistency.

This is what I noticed in the logs abt the replicas,
**Instance 1 and 2:**

> Individual node >>>>com.netflix.eureka.cluster.PeerEurekaNode@472d7c37
> Individual node >>>>http://admin:admin@jhipster-registry-0.jhipster-registry.default.svc.cluster.local:8761/eureka/
> Individual node >>>>com.netflix.eureka.cluster.PeerEurekaNode@d41a9356
> Individual node >>>>http://admin:admin@jhipster-registry-1.jhipster-registry.default.svc.cluster.local:8761/eureka/

**Instance 3:**

> Individual node >>>>com.netflix.eureka.cluster.PeerEurekaNode@6107aa75
> Individual node >>>>http://admin:admin@jhipster-registry-2.jhipster-registry.default.svc.cluster.local:8761/eureka/
> Individual node >>>>com.netflix.eureka.cluster.PeerEurekaNode@472d7c37
> Individual node >>>>http://admin:admin@jhipster-registry-0.jhipster-registry.default.svc.cluster.local:8761/eureka/
> Individual node >>>>com.netflix.eureka.cluster.PeerEurekaNode@d41a9356
> Individual node >>>>http://admin:admin@jhipster-registry-1.jhipster-registry.default.svc.cluster.local:8761/eureka/

After we increase the replica count, instance 1 and 2 are not detecting the new instances as replicas. Changing details in application.yml will get reflected only in new instances, but not in already running ones. 
Though the configuration `(org.springframework.cloud.netflix.eureka.EurekaClientAutoConfiguration$RefreshableEurekaClientConfiguration)` uses  `@RefreshScope`, it needs to be triggered after the property change for the existing instances to get refreshed. 

I am still not convinced with this. Can we just keep it open for 1 or 2 days? I am thinking of writing something in jhipster-registry to make this configMap refresh happens automatically.